astropy/coordinates/baseframe.py

```diff
@@ -617,6 +612,31 @@ def data(self):
                              'data'.format(repr(self)))
         return self._data
 
+    def update_representation_data(self, representation_data):
+        """
+        Update the data in this frame.
+
+        .. warning::
+            This is a low-level in-place operation on this frame. A safer
```

Comment:
The third sentence `this method ...` is a bit ambiguous as which method `this` refers to.  I would suggest:
```
This is a low-level in-place operation on this frame that is provided for 
applications where speed is required.  No validity or consistency checks are performed. 
A safer method of changing the data of a frame is ..., 
which returns a copy of the frame with the new data.
```